File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 4/Exercicio 4.js
class Pessoa{
    constructor(nome, idade){
        this.nome = nome
        this.idade = idade
    }

    dizerNome(){
        return 'Meu nome é: '+this.nome
    }
}

class Aluno extends Pessoa {
    constructor(nome, idade, matricula){
        super(nome, idade)
        this.matricula = matricula
    }

    dizerMatricula(){
        return 'Minha matricula é: '+this.matricula
    }

    dizerNomeMatricula(){
        console.log (super.dizerNome()+' e '+this.dizerMatricula())
    }
}

let al = new Aluno('Maria Antonieta', 25, '20198745')
console.log(al.dizerNome())
console.log(al.dizerMatricula())
al.dizerNomeMatricula()

File: 04 - ORIENTAÇÃO A OBJETOS/Python/Exercicio 4/Exercicio 4.py
class Pessoa:
    def __init__(self, nome, idade):
        self.nome = nome
        self.idade = idade
    
    def dizerNome(self):
        return 'Meu nome é: {}'.format(self.nome)

class Aluno(Pessoa):
    def __init__(self, nome, idade, matricula):
        super().__init__(nome, idade)
        self.matricula = matricula
    
    def dizerMatricula(self):
        return 'Minha matrícula é: {}'.format(self.matricula)
    
    def dizerNomeMatricula(self):
        print('{} e {}'.format(self.dizerNome(), self.dizerMatricula()))

aluno = Aluno('Ygor', 23, '2017586')
print(aluno.dizerNome())
print(aluno.dizerMatricula())
aluno.dizerNomeMatricula()
